bundle/config/validate/validate_sync_patterns.go

```diff
@@ -49,7 +50,11 @@ func checkPatterns(patterns []string, path string, rb bundle.ReadOnlyBundle) (di
 
 	for i, pattern := range patterns {
 		index := i
-		p := pattern
+		fullPattern := pattern
+		// If the pattern is negated, strip the negation prefix
+		// and check if the pattern matches any files.
+		// If it doesn't, it means that negation pattern is not matching any files.
```

Comment:
Doesn't this mean that the negation matches all files? If the pattern is `!p` and `p` matches 0 files, then `!p` should match all files right?

I'm not sure I understand the intention behind this change or if I'm overlooking something.

Reply:
Negation in gitignore syntax means "don't look at this path" pretty much and not the opposite set of match expression. So if `p` matches nothing it's useless negation, but if it matches 2 files let's say, it means: do not include these files into result set

Reply:
Got it. This makes sense then. Could you also add this in the comment above for future readers of the code?